scipy/optimize/_linprog_util.py

```diff
@@ -257,6 +263,31 @@ def _clean_inputs(c, A_ub=None, b_ub=None, A_eq=None, b_eq=None, bounds=None):
             "numerical values, each representing the right hand side of an "
             "equality constraints (row) in A_eq")
 
+    if x0 is not None:
+        try:
+            try:
+                x0 = np.asarray(x0, dtype=float).copy().squeeze()
+            except BaseException:
```

Comment:
I think it's cleaner to just propagate the initial error, so remove this try-except.

Reply:
and `copy` is not needed it seems. or is `x0` modified by some other function?

Reply:
I see this is just copying the pattern that was already here. that's quite odd though, the whole thing is almost unreadable. I suggest stripping down the rest as well.

Reply:
I agree with Matt. It should be cleaned up, but only after the changes in #9671 are addressed.